man/fi_mr.3.md

```diff
@@ -351,6 +357,27 @@ memory region is based on the bitwise OR of the following flags.
   through which the MR is accessed be created with the FI_RMA_EVENT
   capability.
 
+## fi_mr_refresh
+
+The use of this call is required to notify the provider of any change
+to the physical pages backing a registered memory region if the
+FI_MR_MMU_NOTIFY mode bit has been set.  This call informs the provider
+that the page table entries associated with the region may have been
+modified, and the provider should verify and update the registered
+region accordingly.  The iov parameter is optional and may be used
+to specify which portions of the registered region requires updating.
+If provider, providers are only guaranteed to update the specified
+address ranges.
+
+The refresh operation has the effect of disabling and re-enabling
+access to the registered region.  Any operations from peers that attempt
+to access the region will fail while the refresh is occurring.
+Additionally, attempts to access the region by the local process
+through libfabric APIs may result in a page fault or other fatal operation.
+
+The fi_mr_refresh call is only needed if the physical pages might have
+been updated after the memory region was created.
+
```

Comment:
@shefty 

So, here is a question. Is FI_MR_MMU_NOTIFY exclusive w.r.t. FI_MR_PROV_KEY? I'm asking because a fi_mr_refresh might cause a new registration key to be created by a provider.

Based on my own interpretation of this, if FI_MR_PROV_KEY is returned by the provider, and FI_MR_MMU_NOTIFY is also given -- then after a successful fi_mr_refresh, applications should exchange the new keys because the fi_mr_key() and fi_mr_desc() values may have changed.